### PR TITLE
Updated genesis null model

### PR DESCRIPTION
**Pull request by amstilp**
This branch updates analysis_pipeline for null model changes in the GENESIS package. In particular:
* Remove code to create and save the model string, since this is now done by GENESIS. One side effect is that the model string is only printed after fitting the null model, instead of before.
* Name covariance matrices before calling `fitNullModel` so that they have the proper names in the GENESIS model string
* Update null model and analysis reports for changes to the null model format in GENESIS
* Update null model test data. As part of this, I added a file showing the commands I ran to generate the test data in case we need to update it again in the future. If we update other test data, we can add those commands too.
* Update the `install_packages.R` script to point at the updated version of GENESIS

### CODE REVIEW

File: R/null_model.R [hunk 1]
```diff
@@ -56,20 +56,14 @@ if (family == "binomial") {
 
 # kinship matrix or GRM
 grm <- getGRM(config, sample.id)
-
-# print model
-random <- if (!is.na(config["relatedness_matrix_file"])) "relatedness" else NULL
-model.string <- modelString(outcome, covars, random, group.var)
-message("Model: ", model.string)
-message(length(sample.id), " samples")
```
Comment (mconomos) [inline R/null_model.R:64]: @amstilp might want to put a message back in about sample size; I think that's useful here

File: TopmedPipeline/inst/rmd/null_model_report.Rmd [hunk 2]
```diff
@@ -369,7 +372,7 @@ if (!is.null(group_var)) {
 
 ## Adjusted phenotype
```
Comment (mconomos) [inline TopmedPipeline/inst/rmd/null_model_report.Rmd:373]: @amstilp I'd like to make a couple changes to this section of the report:

Rename to "Phenotype adjustment"

File: TopmedPipeline/inst/rmd/null_model_report.Rmd [hunk 3]
```diff
@@ -369,7 +372,7 @@ if (!is.null(group_var)) {
...
```
Comment (mconomos) [inline TopmedPipeline/inst/rmd/null_model_report.Rmd:374]: "The Cholesky residuals (`resid.cholesky`) are adjusted for fixed effect covariates,
random effects, and heterogeneous residual variances (if applicable). They are a transformation of the marginal residuals using the estimated model covariance structure such that they should be uncorrelated and follow a standard multivariate Normal distribution with mean 0 and identity covariance matrix asymptotically. Linear regression of the Cholesky residual vector on an equivalently transformed genotype vector provides the same estimates as fitting the full GLS model (by the Frisch-Waugh-Lovell theorem)."

*move the code for the `PY` histogram up here, but replace `PY` with `resid.cholesky`

The square of these values should have a trend line near the y = 1 line, indicating homoskedasticity after adjustment. Deviation from this trend could indicate that the variance model is inadequate.

*the plot(s) of `resid.cholesky^2`

File: TopmedPipeline/inst/rmd/null_model_report.Rmd [hunk 4]
```diff
@@ -405,13 +408,13 @@ if (is_binary) {
...
   } else {
-    extra_text <- "A histogram of `PY` values is shown below."
+    extra_text <- "A histogram of `resid.PY` values is shown below."
   }
   if(!is_invnorm) {
     extra_text <- paste(
       extra_text,
       "\n\nIt may be problematic for association testing if this distribution is far
-      from Gaussian. This problem can be alleviated by rank normalizing `PY` and
+      from Gaussian. This problem can be alleviated by rank normalizing `resid.PY` and
```
Comment (mconomos) [inline TopmedPipeline/inst/rmd/null_model_report.Rmd:417]: ...by rank normalizing `resid.marginal` and re-fitting...

**Review by smgogarten — APPROVED** [review]
All looks good to me, thanks!

File: R/null_model.R (hunk 1 above)
Comment (amstilp) [reply R/null_model.R:64]: Good point, I added that part of the message back in.

File: TopmedPipeline/inst/rmd/null_model_report.Rmd (hunk 4 above)
Comment (amstilp) [reply TopmedPipeline/inst/rmd/null_model_report.Rmd:417]: Fixed

File: TopmedPipeline/inst/rmd/null_model_report.Rmd (hunk 3 above)
Comment (amstilp) [reply TopmedPipeline/inst/rmd/null_model_report.Rmd]: I think I got all these changes implemented. Thanks for chatting about them.

**Pull request merged**